std/json.d

```diff
@@ -346,10 +346,11 @@ struct JSONValue
             type_tag = JSON_TYPE.STRING;
             store.str = arg;
         }
-        else static if(is(T : char[]))
+        else static if(is(T : const char[]) || is(T : const wchar[]) || is(T : const dchar[]))
         {
+            import std.utf : byUTF;
             type_tag = JSON_TYPE.STRING;
-            store.str = arg.idup;
+            store.str = byUTF!char(arg).array.idup;
```

Comment:
@ZombineDev `byUTF` is pure, but `std.array.array(Range)` is not, because (I suppose) it accepts non-pure ranges. I guess dmd wasn't able to deduce pureness here.